pkg/virt-launcher/virtwrap/network/common_test.go

```diff
@@ -68,37 +69,47 @@ var _ = Describe("Common Methods", func() {
 		})
 	})
 	Context("GetAvailableAddrsFromCIDR function", func() {
-		It("Should return 2 addresses", func() {
-			networkHandler := NetworkUtilsHandler{}
-			gw, vm, err := networkHandler.GetHostAndGwAddressesFromCIDR("10.0.0.0/30")
-			Expect(err).ToNot(HaveOccurred())
-			Expect(gw).To(Equal("10.0.0.1/30"))
-			Expect(vm).To(Equal("10.0.0.2/30"))
-		})
-		It("Should return 2 IPV6 addresses", func() {
-			networkHandler := NetworkUtilsHandler{}
-			gw, vm, err := networkHandler.GetHostAndGwAddressesFromCIDR("fd10:0:2::/120")
-			Expect(err).ToNot(HaveOccurred())
-			Expect(gw).To(Equal("fd10:0:2::1/120"))
-			Expect(vm).To(Equal("fd10:0:2::2/120"))
-		})
-		It("Should fail when the subnet is too small", func() {
-			networkHandler := NetworkUtilsHandler{}
-			_, _, err := networkHandler.GetHostAndGwAddressesFromCIDR("10.0.0.0/31")
-			Expect(err).To(HaveOccurred())
-		})
-		It("Should fail when the IPV6 subnet is too small", func() {
-			networkHandler := NetworkUtilsHandler{}
-			_, _, err := networkHandler.GetHostAndGwAddressesFromCIDR("fd10:0:2::/127")
-			Expect(err).To(HaveOccurred())
-		})
-	})
-	Context("GenerateRandomMac function", func() {
-		It("should return a valid mac address", func() {
-			networkHandler := NetworkUtilsHandler{}
-			mac, err := networkHandler.GenerateRandomMac()
-			Expect(err).ToNot(HaveOccurred())
-			Expect(strings.HasPrefix(mac.String(), "02:00:00")).To(BeTrue())
+		var testParams = map[string]struct {
+			cidr       string
+			smallCider string
+			samples    map[string]string
+		}{
+			"IPV4": {
+				"10.0.2.0/30",
+				"10.0.0.0/31",
+				map[string]string{
+					"GW": "10.0.2.1/30",
+					"VM": "10.0.2.2/30"}},
+			"IPV6": {
+				api.DefaultVMIpv6CIDR,
+				"fd10:0:2::/127",
+				map[string]string{
+					"GW": "fd10:0:2::1/120",
+					"VM": "fd10:0:2::2/120"}},
+		}
+
+		for ipVersion, currentCidr := range testParams {
+			It(fmt.Sprintf("Should return %d %s addresses",len(currentCidr.samples), ipVersion), func() {
+				networkHandler := NetworkUtilsHandler{}
+				gw, vm, err := networkHandler.GetHostAndGwAddressesFromCIDR(currentCidr.cidr)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(gw).To(Equal(currentCidr.samples["GW"]))
+				Expect(vm).To(Equal(currentCidr.samples["VM"]))
+			})
+
+			It(fmt.Sprintf("Should fail when the %s subnet is too small", ipVersion), func() {
+				networkHandler := NetworkUtilsHandler{}
+				_, _, err := networkHandler.GetHostAndGwAddressesFromCIDR(testParams[ipVersion].smallCider)
+				Expect(err).To(HaveOccurred())
+			})
+		}
+		Context("GenerateRandomMac function", func() {
```

Comment:
Please move it outside of the "GetAvailableAddrsFromCIDR function" Context.

Reply:
I reverted this for loop, I will continue with this in a new PR